view/frontend/web/js/track-checkout.js

```diff
@@ -0,0 +1,70 @@
+/* Track checkout funnel transition in magento's default checkout */
+
+// TODO: support logged in user
+
+var currentStep = '';
+
+function waitFor(condition, doWhenReady) {
```

Comment:
curious: doesnt jquery have its own way to wait for conditions? whats thwe advantage of this ^

Reply:
hm I don't think jquery has this kind of polling wait

Reply:
I think we should add a new argurment to specify the number of retry attempts to check the condition. If the condition always returns false, we might end up running this check forever in the checkout page. We should also assign a default value for that argument like 50.